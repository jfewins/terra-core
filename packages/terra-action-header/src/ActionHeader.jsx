import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './ActionHeader.scss';

const cx = classNames.bind(styles);

const propTypes = {
 /*
 * Content to be displayed as the name
 */
  name: PropTypes.string,
};

const defaultProps = {
  name: 'default',
};

const ActionHeader = ({ name, ...customProps }) => {
  const ActionHeaderClassNames = cx([
    'action-header',
    customProps.className,
  ]);

  return (<div {...customProps} className={ActionHeaderClassNames}>{name}</div>);
};

ActionHeader.propTypes = propTypes;
ActionHeader.defaultProps = defaultProps;

export default ActionHeader;
