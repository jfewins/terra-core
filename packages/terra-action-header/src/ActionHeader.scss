:local {
  .action-header {
    /* TODO: Add styles */
  }
}
