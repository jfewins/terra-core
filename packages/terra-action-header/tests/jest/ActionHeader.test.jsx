import React from 'react';
import ActionHeader from '../../src/ActionHeader';

describe('ActionHeader', () => {
  // Snapshot Tests
  it('should render a default action header', () => {
    const actionHeader = render(<ActionHeader />);
    expect(actionHeader).toMatchSnapshot();
  });

  it('should render an action header with title', () => {
    const actionHeader = render(<ActionHeader title="Action Header" />);
    expect(actionHeader).toMatchSnapshot();
  });

  it('should render an action header with back button and title', () => {
    const actionHeader = render(<ActionHeader onBack={() => {}} title="Action Header" />);
    expect(actionHeader).toMatchSnapshot();
  });

  it('should render an action header with close button and title', () => {
    const actionHeader = render(<ActionHeader onClose={() => {}} title="Action Header" />);
    expect(actionHeader).toMatchSnapshot();
  });

  it('should render an action header with back and close buttons and title', () => {
    const actionHeader = render(<ActionHeader onBack={() => {}} onClose={() => {}} title="Action Header" />);
    expect(actionHeader).toMatchSnapshot();
  });
  // // Prop Tests
  // it('should use the default value when no value is given', () => {
  //   const wrapper = shallow(defaultRender);
  //   expect(wrapper.find('.action-header').text()).toEqual('default');
  // });
  //
  // // Structure Tests
  // it('should have the class action-header', () => {
  //   const wrapper = shallow(defaultRender);
  //   expect(wrapper.prop('className')).toContain('action-header');
  // });
});
