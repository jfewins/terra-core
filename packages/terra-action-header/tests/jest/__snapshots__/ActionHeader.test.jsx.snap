// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionHeader should render a default component 1`] = `
<div
  className="action-header"
>
  default
</div>
`;
